test/program_test.cpp

```diff
@@ -27,4 +30,46 @@ TEST_CASE(program_equality)
     EXPECT(x == y);
 }
 
+TEST_CASE(program_copy)
+{
+    auto create_program_1 = [] {
+        migraphx::program p;
+        migraphx::shape s{migraphx::shape::float_type, {3, 4, 5}};
+        std::vector<float> data(3 * 4 * 5);
+        std::iota(data.begin(), data.end(), 1.0f);
+        auto l2  = p.add_literal(migraphx::literal(s, data));
+        auto p1  = p.add_parameter("x", s);
+        auto po  = p.add_outline(s);
+        auto sum = p.add_instruction(migraphx::op::add{}, l2, p1);
+        p.add_instruction(migraphx::op::mul{}, sum, po);
+
+        return p;
+    };
+
+    {
+        auto p1 = create_program_1();
+        auto p2 = p1;
+        p2.compile(migraphx::cpu::target{});
+        EXPECT(p1 == p2);
```

Comment:
We should have a test that `p1 != p2` after compiling.

Reply:
But this should be `false` since `op::add` will get lowered to `cpu::add`.

Reply:
rename the name of the cpu operator, now it works as expected.